### PR TITLE
Requestify Operator Decl Attachment

### PR DESCRIPTION
**Pull request by CodaFi**
Also perform a small amount of cleanup around the printing.  I'd love to revisit the tangled diagnostics logic in here once we have a better story for how diagnostics-emitting requests are going to work,

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

File: lib/Sema/TypeCheckDecl.cpp [hunk 1]
```diff
@@ -3868,10 +3724,9 @@ void TypeChecker::validateDecl(ValueDecl *D) {
 
     DeclValidationRAII IBV(FD);
 
-    // Bind operator functions to the corresponding operator declaration.
-    if (FD->isOperator())
-      bindFuncDeclToOperator(*this, FD);
-
+    // Force computing the operator decl in case it emits diagnostics.
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDecl.cpp]: We do request forcing in typeCheckDecl(). No need to trigger this for decls in secondary files for SILGen's sake.

Also further down in the same switch case there is this block: `// Member functions need some special validation logic.`. I think this check should go into the request or even just typeCheckDecl() itself.

File: lib/Sema/TypeCheckDecl.cpp [hunk 2]
```diff
@@ -3429,150 +3429,6 @@ static void validateTypealiasType(TypeChecker &tc, TypeAliasDecl *typeAlias) {
   typeAlias->setUnderlyingType(typeAlias->getUnderlyingTypeLoc().getType());
 }
 
-
-/// Bind the given function declaration, which declares an operator, to
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDecl.cpp]: To make the diff smaller and easier to read, can you leave the ::evaluate() method in the same location as this function used to be? We can re-organize TypeCheckDecl.cpp later.

File: lib/Sema/TypeCheckDecl.cpp [hunk 3]
```diff
@@ -4212,6 +4067,169 @@ EmittedMembersRequest::evaluate(Evaluator &evaluator,
   return CD->getMembers();
 }
 
+Optional<OperatorDecl *> swift::FunctionOperatorRequest::getCachedResult() const {
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDecl.cpp]: We've been defining getCachedResult()/cacheResult() in TypeCheckRequests.cpp instead, but in this case if you use internal caching you won't have to define them at all.

**Review by slavapestov — COMMENTED** [review]
Awesome!

**Comment by slavapestov** [comment]
@CodaFi For now, I think emitting diagnostics as a side effect of evaluating a request is totally fine. I wouldn't worry about trying to change the design to handle this in a better way just yet.

In the long term, when we want to re-parse files and invalidate cached requests, we will need to know which requests produced which diagnostics, though. But one step at a time...

**Comment by CodaFi** [comment]
Going to apply this feedback and reopen this and the other one - I accidentally created these branches against this repo instead of my fork.

**Pull request closed without merging**